### PR TITLE
create user model

### PR DESCRIPTION
**Pull request by alexdavcudriz**
## Summary
- User model was created.
- Book exist in user exception was created.
## Trello Card
https://trello.com/c/YUAhZXyn/19-crear-modelo-de-user

### CODE REVIEW

File: src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package wolox.training.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.PRECONDITION_FAILED, reason = "Repeated book")
+public class BookAlreadyOwnedException extends Exception {
```
Comment (jdhiguita) [inline src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java]: RuntimeException instead Exception

File: src/main/java/wolox/training/models/User.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @Column(nullable = false)
+    private String username;
+    @NotNull
+    @Column(nullable = false)
+    private String name;
+    @NotNull
+    @Column(nullable = false)
+    private LocalDate birthdate;
+    @ManyToMany
+    private List<Book> books = Collections.emptyList();
```
Comment (jdhiguita) [inline src/main/java/wolox/training/models/User.java:38]: When adding a many to many relation we need the other part of the relation. Now we should add the field into the book model.

  @ManyToMany (mappedBy = "books")
  private List <User> user;
Remember to add its respective get as you did in this collection.

Additional a good practice is add {CascadeType.MERGE, CascadeType.REFRESH}), although it is not mandatory for this card

File: src/main/java/wolox/training/models/User.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public List<Book> getBooks() {
+        return Collections.unmodifiableList(books);
```
Comment (jdhiguita) [inline src/main/java/wolox/training/models/User.java]: Add cast of List<Book> for this return

**Review by jdhiguita — COMMENTED** [review]
Good job! Some comments for you

File: src/main/java/wolox/training/models/User.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package wolox.training.models;
+
+import com.sun.istack.NotNull;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+import javax.persistence.*;
```
Comment (facucompe) [inline src/main/java/wolox/training/models/User.java]: Remove wildcards in imports. https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports

File: src/main/java/wolox/training/controllers/BookController.java [hunk 5]
```diff
@@ -41,12 +41,7 @@ public Iterable findAll() {
      */
     @GetMapping("/{id}")
     public Book findOne(@PathVariable Long id) {
-        try {
-            return bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
-        } catch (BookNotFoundException e) {
-            throw new ResponseStatusException(
-                    HttpStatus.NOT_FOUND, "Book not found", e);
-        }
+        return bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
```
Comment (jdhiguita) [inline src/main/java/wolox/training/controllers/BookController.java:44]: User services instead repositories in controllers, the card not mention that but is a good practice

**Review by jdhiguita — COMMENTED** [review]
Last comment

**Review by jdhiguita — APPROVED** [review]

**Review by facucompe — APPROVED** [review]

**Pull request merged**